das/daser.go

```diff
@@ -15,28 +15,63 @@ import (
 	"github.com/celestiaorg/celestia-node/share"
 )
 
+// TODO(@team): Should we DI this?
 var log = logging.Logger("das")
 
-// TODO: parameters needs performance testing on real network to define optimal values
-const (
-	//  samplingRange is the maximum amount of headers processed in one job.
-	samplingRange = 100
+type dasingParams struct {
+	samplingRange     uint
+	concurrencyLimit  uint
+	bgStoreInterval   time.Duration
+	priorityQueueSize uint
+	genesisHeight     uint
+}
 
-	// concurrencyLimit defines the maximum amount of sampling workers running in parallel.
-	concurrencyLimit = 16
+func defaultDasingParams() dasingParams {
+	return dasingParams{
+		samplingRange:     100,
+		concurrencyLimit:  16,
+		bgStoreInterval:   10 * time.Minute,
+		priorityQueueSize: 16 * 4,
+		genesisHeight:     1,
+	}
+}
 
-	// backgroundStoreInterval is the period of time for background checkpointStore to perform a checkpoint backup.
-	backgroundStoreInterval = 10 * time.Minute
+type Option func(*DASer)
 
-	// priorityQueueSize defines the size limit of the priority queue
-	priorityQueueSize = concurrencyLimit * 4
+func WithParamSamplingRange(samplingRange uint) Option {
+	return func(d *DASer) {
+		d.params.samplingRange = samplingRange
+	}
+}
 
-	// genesisHeight is the height sampling will start from
-	genesisHeight = 1
-)
+func WithParamConcurrencyLimit(concurrencyLimit uint) Option {
+	return func(d *DASer) {
+		d.params.concurrencyLimit = concurrencyLimit
+	}
+}
+
+func WithParamBackgroundStoreInterval(bgStoreInterval time.Duration) Option {
+	return func(d *DASer) {
+		d.params.bgStoreInterval = bgStoreInterval
+	}
+}
+
+func WithParamPriorityQueueSize(priorityQueueSize uint) Option {
+	return func(d *DASer) {
+		d.params.priorityQueueSize = priorityQueueSize
+	}
+}
+
+func WithParamGenesisHeight(genesisHeight uint) Option {
```

Comment:
I don't think these options, nor `Option` belong in the `das/` pkg

This should all live at `nodebuilder/daser` since they're configuration options.

Did you already discuss the placement with @Wondertan ?

Reply:
They should be here. These are options for the DASer, not the module.

Reply:
To add to @Wondertan's comment, it is with these `Option`s that the `nodebuilder/daser` will be able to "configure" the `das/daser`.